diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/common/rfb/CSecurityVncAuth.cxx vnc-4_1_3-unixsrc-4.1.3/common/rfb/CSecurityVncAuth.cxx
--- vnc-4_1_3-unixsrc-4.1.3.orig/common/rfb/CSecurityVncAuth.cxx	2025-11-14 08:36:21.720832585 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/common/rfb/CSecurityVncAuth.cxx	2025-11-14 08:39:37.092817004 -0500
@@ -28,9 +28,13 @@
 #include <rfb/Password.h>
 #include <rfb/CSecurityVncAuth.h>
 #include <rfb/util.h>
+#ifdef __cplusplus
 extern "C" {
+#endif
 #include <rfb/d3des.h>
+#ifdef __cplusplus
 }
+#endif
 
 
 using namespace rfb;
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/common/rfb/Password.cxx vnc-4_1_3-unixsrc-4.1.3/common/rfb/Password.cxx
--- vnc-4_1_3-unixsrc-4.1.3.orig/common/rfb/Password.cxx	2025-11-14 08:36:21.720832585 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/common/rfb/Password.cxx	2025-11-14 08:38:41.000823359 -0500
@@ -21,9 +21,13 @@
 //
 
 #include <string.h>
+#ifdef __cplusplus
 extern "C" {
+#endif
 #include <rfb/d3des.h>
+#ifdef __cplusplus
 }
+#endif
 #include <rdr/types.h>
 #include <rdr/Exception.h>
 #include <rfb/Password.h>
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/common/rfb/SSecurityVncAuth.cxx vnc-4_1_3-unixsrc-4.1.3/common/rfb/SSecurityVncAuth.cxx
--- vnc-4_1_3-unixsrc-4.1.3.orig/common/rfb/SSecurityVncAuth.cxx	2025-11-14 08:36:21.724832585 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/common/rfb/SSecurityVncAuth.cxx	2025-11-14 08:37:27.816829473 -0500
@@ -31,9 +31,13 @@
 #include <rfb/Exception.h>
 #include <string.h>
 #include <stdio.h>
+#ifdef __cplusplus
 extern "C" {
+#endif
 #include <rfb/d3des.h>
+#ifdef __cplusplus
 }
+#endif
 
 
 using namespace rfb;
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/vncExtInit.cc vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/vncExtInit.cc
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/vncExtInit.cc	2025-11-14 08:36:21.736832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/vncExtInit.cc	2025-11-14 09:51:35.023140870 -0500
@@ -20,8 +20,12 @@
 
 #include "XserverDesktop.h"
 
+#ifdef __cplusplus
 extern "C" {
+#endif
+#ifndef __cplusplus
 #define class c_class
+#endif
 #define NEED_EVENTS
 #include "X.h"
 #include "Xproto.h"
@@ -32,9 +36,13 @@
 #define _VNCEXT_SERVER_
 #define _VNCEXT_PROTO_
 #include "vncExt.h"
+#ifndef __cplusplus
 #undef class
+#endif
 #undef xalloc
+#ifdef __cplusplus
 }
+#endif
 
 #include <rfb/Configuration.h>
 #include <rfb/Logger_stdio.h>
@@ -52,7 +60,9 @@
 #include "vncHooks.h"
 #include "vncExtInit.h"
 
+#ifdef __cplusplus
 extern "C" {
+#endif
 
   extern void vncExtensionInit();
   static void vncResetProc(ExtensionEntry* extEntry);
@@ -67,7 +77,9 @@
 
   extern Selection *CurrentSelections;
   extern int NumCurrentSelections;
+#ifdef __cplusplus
 }
+#endif
 
 using namespace rfb;
 
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/vncHooks.cc vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/vncHooks.cc
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/vncHooks.cc	2025-11-14 08:36:21.736832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/vncHooks.cc	2025-11-14 10:49:45.253837769 -0500
@@ -20,9 +20,13 @@
 #include "XserverDesktop.h"
 #include "vncHooks.h"
 
+#ifdef __cplusplus
 extern "C" {
+#endif
+#ifndef __cplusplus
 #define class c_class
 #define private c_private
+#endif
 #include "scrnintstr.h"
 #include "windowstr.h"
 #include "gcstruct.h"
@@ -41,9 +45,13 @@
   (((mfbPrivGCPtr)((gc)->devPrivates[mfbGCPrivateIndex].ptr))->pCompositeClip)
 #endif
 
+#ifndef __cplusplus
 #undef class
 #undef private
+#endif
+#ifdef __cplusplus
 }
+#endif
 
 #include "RegionHelper.h"
 
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/vncHooks.h vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/vncHooks.h
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/vncHooks.h	2025-11-14 08:36:21.736832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/vncHooks.h	2025-11-14 10:00:44.082809578 -0500
@@ -18,9 +18,13 @@
 #ifndef __VNCHOOKS_H__
 #define __VNCHOOKS_H__
 
+#ifdef __cplusplus
 extern "C" {
+#endif
 #include <screenint.h>
   extern Bool vncHooksInit(ScreenPtr pScreen, XserverDesktop* desktop);
+#ifdef __cplusplus
 }
+#endif
 
 #endif
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/xf86vncModule.cc vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/xf86vncModule.cc
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/xf86vncModule.cc	2025-11-14 08:36:21.736832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/xf86vncModule.cc	2025-11-14 10:48:15.649807104 -0500
@@ -22,17 +22,23 @@
 #include <rfb/Logger_stdio.h>
 #include <rfb/LogWriter.h>
 
+#ifdef __cplusplus
 extern "C" {
+#endif
+#ifndef __cplusplus
 #define class c_class
 #define private c_private
 #define bool c_bool
 #define new c_new
+#endif
 #include "xf86.h"
 #include "xf86Module.h"
+#ifndef __cplusplus
 #undef class
 #undef private
 #undef bool
 #undef new
+#endif
 
 using namespace rfb;
 
@@ -94,4 +100,6 @@
 }
 
 #endif /* XFree86LOADER */
+#ifdef __cplusplus
 }
+#endif
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/XserverDesktop.cc vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/XserverDesktop.cc
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/XserverDesktop.cc	2025-11-14 08:36:21.732832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/XserverDesktop.cc	2025-11-14 09:58:50.586872733 -0500
@@ -36,9 +36,13 @@
 #include "XserverDesktop.h"
 #include "vncExtInit.h"
 
+#ifdef __cplusplus
 extern "C" {
+#endif
+#ifndef __cplusplus
 #define public c_public
 #define class c_class
+#endif
 
   // windowTable is in globals.h in XFree 4, but not in XFree 3 unfortunately
 extern WindowPtr *WindowTable;
@@ -56,9 +60,13 @@
 #include "windowstr.h"
 #define XK_CYRILLIC
 #include "keysym.h"
+#ifndef __cplusplus
 #undef public
 #undef class
+#endif
+#ifdef __cplusplus
 }
+#endif
 
 using namespace rfb;
 using namespace network;
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/XserverDesktop.h vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/XserverDesktop.h
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/XserverDesktop.h	2025-11-14 08:36:21.732832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/XserverDesktop.h	2025-11-14 10:48:48.521819576 -0500
@@ -29,12 +29,20 @@
 #include <rfb/VNCServerST.h>
 #include <rdr/SubstitutingInStream.h>
 
+#ifdef __cplusplus
 extern "C" {
+#endif
+#ifndef __cplusplus
 #define class c_class;
+#endif
 #include <scrnintstr.h>
 #include <os.h>
+#ifndef __cplusplus
 #undef class
+#endif
+#ifdef __cplusplus
 }
+#endif
 
 namespace rfb {
   class VNCServerST;
diff -Naur vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/Xvnc/xvnc.cc vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/Xvnc/xvnc.cc
--- vnc-4_1_3-unixsrc-4.1.3.orig/unix/xc/programs/Xserver/vnc/Xvnc/xvnc.cc	2025-11-14 08:36:21.732832584 -0500
+++ vnc-4_1_3-unixsrc-4.1.3/unix/xc/programs/Xserver/vnc/Xvnc/xvnc.cc	2025-11-14 08:45:04.224758273 -0500
@@ -33,11 +33,15 @@
 #include <network/TcpSocket.h>
 #include "vncExtInit.h"
 
+#ifdef __cplusplus
 extern "C" {
+#endif
+#ifndef __cplusplus
 #define class c_class
 #define public c_public
 #define xor c_xor
 #define and c_and
+#endif
 #ifdef WIN32
 #include <X11/Xwinsock.h>
 #endif
@@ -73,10 +77,12 @@
 #include "keysym.h"
   extern int defaultColorVisualClass;
   extern char buildtime[];
+#ifndef __cplusplus
 #undef class
 #undef public
 #undef xor
 #undef and
+#endif
 #ifndef VNC_USE_FB
   extern Bool cfb16ScreenInit(ScreenPtr, pointer, int, int, int, int, int);
   extern Bool cfb32ScreenInit(ScreenPtr, pointer, int, int, int, int, int);
@@ -87,7 +93,9 @@
   extern void cfb32GetImage(DrawablePtr, int, int, int, int, unsigned int,
                             unsigned long, char*);
 #endif
+#ifdef __cplusplus
 }
+#endif
 
 #define XVNCVERSION "Free Edition 4.1.3"
 #define XVNCCOPYRIGHT ("Copyright (C) 2002-2008 RealVNC Ltd.\n" \
@@ -109,9 +117,13 @@
 
 #ifdef RANDR
 
+#ifdef __cplusplus
 extern "C" {
+#endif
 #include <randrstr.h>
+#ifdef __cplusplus
 }
+#endif
 
 #define RR_MAX_SCREEN_SIZES     8
 typedef struct { int width, height; } rrScreenSize;
@@ -179,7 +191,9 @@
 }
 
 
+#ifdef __cplusplus
 extern "C" {
+#endif
 
   /* ddxInitGlobals - called by |InitGlobals| from os/util.c in XOrg */
   void ddxInitGlobals(void)
@@ -238,7 +252,9 @@
             "Parameter names are case-insensitive.  The parameters are:\n\n");
     rfb::Configuration::listParams(79, 14);
   }
+#ifdef __cplusplus
 }
+#endif
 
 static bool displayNumFree(int num)
 {
@@ -962,7 +978,9 @@
 } /* end InitOutput */
 
 #ifdef DPMSExtension
+#ifdef __cplusplus
 extern "C" {
+#endif
 #if NeedFunctionPrototypes
   void DPMSSet(CARD16 level)
 #else
@@ -977,7 +999,9 @@
   {
     return FALSE;
   }
+#ifdef __cplusplus
 }
+#endif
 #endif
 
 /* this is just to get the server to link on AIX */
