diff -Naur vnc4-4.1.1+X4.3.0.orig/unix/xc/programs/Xserver/hw/xfree86/fbdevhw/fbdevhw.c vnc4-4.1.1+X4.3.0.orig_monopatch/unix/xc/programs/Xserver/hw/xfree86/fbdevhw/fbdevhw.c
--- vnc4-4.1.1+X4.3.0.orig/unix/xc/programs/Xserver/hw/xfree86/fbdevhw/fbdevhw.c	2002-11-25 09:05:00.000000000 -0500
+++ vnc4-4.1.1+X4.3.0.orig_monopatch/unix/xc/programs/Xserver/hw/xfree86/fbdevhw/fbdevhw.c	2025-10-23 07:26:04.754999736 -0400
@@ -14,7 +14,17 @@
 #include "fbdevhw.h"
 #include "fbpriv.h"
 
-#include "asm/page.h"	/* #define for PAGE_* */
+/* Added check so that asm/page.h is not included for hppa arch */
+/* Ola Lundqvist, to work around Debian Bug#485958 & Bug#486104*/
+#if !defined(__hppa__) && !defined(__mips__) && !defined(__alpha__) && !defined(__x86_64__)
+# include "asm/page.h"	/* #define for PAGE_* */
+#endif
+
+#include <unistd.h>
+
+#ifndef PAGE_MASK
+#define PAGE_MASK (~(sysconf(_SC_PAGESIZE) - 1))
+#endif
 
 #include "globals.h"
 #define DPMS_SERVER
