--- vnc4-4.1.1+X4.3.0_gismo-4/unix/vncviewer/CConn.cxx	2006-11-07 19:43:39.000000000 +0100
+++ vnc4-4.1.1+X4.3.0_gismo-5/unix/vncviewer/CConn.cxx	2006-11-07 19:43:05.000000000 +0100
@@ -91,7 +91,7 @@
       int popup = popupXDialog;
       if (!popup) {
         /* Get server */
-        fprintf(stderr, "Server: ");
+        fprintf(stderr, "\nServer: ");
         vncServerName = new char[128];
 	if(fgets(vncServerName, 128, stdin)) {
 	  size_t len = strlen(vncServerName);
@@ -232,7 +232,7 @@
   if (!popup) {
     if (user) {
       /* Get username */
-      fprintf(stderr, "Username: ");
+      fprintf(stderr, "\nUsername: ");
       *user = new char[128];
       fgets(*user, 128, stdin);
       /* Remove \n at the end */
@@ -244,7 +244,7 @@
     } else {
       newio = oldio;
       newio.c_lflag &= ~ECHO;
-      fprintf(stderr, "Password: ");
+      fprintf(stderr, "\nPassword: ");
       /* Echo off */
       if (tcsetattr (fileno (stdin), TCSAFLUSH, &newio) != 0)
 	popup = 1;
